src/Artifacts.jl

```diff
@@ -326,7 +326,7 @@ function download_artifact(
             if isa(err, InterruptException)
                 rethrow(err)
             end
-            return false
+            return err
```

Comment:
Should mention in the docstring that this can return an exception object

Reply:
This is breaking, right? It's perfectly legal for us to do, since this isn't a public function, but we will probably have to fix some packages that currently assume that this returns a `Bool`.

E.g. I think that `JuliaCI/rootfs-images` uses this function and assumes that it returns a `Bool`.

Reply:
Totally. Thanks

Reply:
Ah. Didn't see your message @DilumAluthge 

Yeah, there's also these https://juliahub.com/ui/Search?q=download_artifact%28&type=code&w=true

Reply:
none of which appear to use the output of the function, so that's convenient!